internal/token_test.go

```diff
@@ -169,6 +169,17 @@ func TestFrontmatter(t *testing.T) {
 			`,
 			[]TokenType{FrontmatterFenceToken, TextToken, TextToken, StartTagToken, StartExpressionToken, TextToken, EndExpressionToken, EndTagToken, TextToken, TextToken, TextToken, TextToken, TextToken, FrontmatterFenceToken},
 		},
+		{
+			"function components w/ conditional rendering",
+			`---
+			const getSummary = (key) => {
+				return <>
+					{key && <><span class={` + "`" + `${className} key` + "`" + `}>{key}</span></>}
+				</>;
+			};
+			---`,
+			[]TokenType{FrontmatterFenceToken, TextToken, TextToken, StartTagToken, TextToken, StartExpressionToken, TextToken, StartTagToken, StartTagToken, StartExpressionToken, TextToken, TextToken, EndExpressionToken, StartExpressionToken, TextToken, EndExpressionToken, EndTagToken, EndTagToken, EndExpressionToken, TextToken, EndTagToken, TextToken, TextToken, FrontmatterFenceToken},
```

Comment:
Is this right? Or should it mostly be text (i.e. not try and flag internal expressions like start tags & end tags?